### PR TITLE
Switch to CircleCI 2.0.

### PR DESCRIPTION
**Pull request by felicianotech**
Release note: Migrate to CircleCI 2.0

Hey everyone,

If you don't know, CircleCI 1.0 [is being EOL'd](https://circleci.com/blog/sunsetting-1-0/) at the end of August. I noticed that this project uses CircleCI 1.0 and so in the spirit of open source I wanted to submit a PR to do the migration.

The **TL:DR;** of this PR is that it converts the CircleCI 1.0 configuration file into the equivalent CircleCI 2.0 configuration file. I'd say that the new config file is 99% the same software-wise. Here's the two changes:
- the Yarn version was bumped  from v1.1.0 to the current version on the Docker Library Node images, which is v1.3.
- I split the running of tests and checks from a single, sequential build into 3 separate jobs.

For that latter point, I noticed that if you averaged the CircleCI 1.0 build time for this project for the last 3 passing `master` builds, the average build time is **32m 49s**. I ran this PR on CircleCI 2.0 and the average build time for the last 3 passing builds is **13m 22s**.

Basically, this PR allows this project to migrate to CircleCI 2.0 before the August 2018 EOL of 1.0 as well as decrease build times by 49%.

I'm happy to answer any questions or help with any tweaks to get this PR merged if you choose to do so.

### CODE REVIEW

**Comment by felicianotech** [comment]
Already signed the CLA ^. Maybe there's simply a delay in the system?

**Comment by NTillmann** [comment]
Thanks for doing this!

It seems that the link GItHub has associated with "All checks have passed" defaults to the last job. Is that the expected experience or something we should (re)configure? I guess it would link to a failing job if there was any failing job?

**Comment by felicianotech** [comment]
I believe that some weirdness due to that the check is from my own CircleCI account and not Facebook's.

Merging this PR, or pushing your own branch with the code from someone who's a collaborator on the repo should initiate a new build under Facebook's org.
